### PR TITLE
Switch over to using pytest to run the python tests

### PR DESCRIPTION
**Pull request by greglandrum**
This is way cleaner than using the old self-made solution from before.

Don't be intimidated by the large number of changed files: most of that is removing the old `test_list.py` files.

There's some additional cleanup in here to:
- remove most of the old infrastructure for running python tests
- remove six.py, which wasn't being used anywhere (and shouldn't be used now that we aren't supporting python 2)

### CODE REVIEW

**Comment by ricrogz** [comment]
One more question: how do I do to run a single Python test? I'm trying to run `python3 -m pytest ../rdkit/Chem/UnitTestChem.py` from my build directory, and it fails with (I'm using `RDK_INSTALL_INTREE=OFF`) :
```
_____________________________________________________________________________________________________________________________________ ERROR collecting ../rdkit/Chem/UnitTestChem::py  ______________________________________________________________________________________________________________________________________
ImportError while importing test module '/tmp/rdkit_builder/rdkit/rdkit/Chem/UnitTestChem.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../../../../lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
E   ModuleNotFoundError: No module named 'rdkit.Chem.UnitTestChem'

```

**Comment by greglandrum** [comment]
> One more question: how do I do to run a single Python test? I'm trying to run `python3 -m pytest ../rdkit/Chem/UnitTestChem.py` from my build directory, and it fails with (I'm using `RDK_INSTALL_INTREE=OFF`) :

That's the correct invocation for a single test. I suspect that the problem is due to the INTREE bit. I will take a look at that.

File: rdkit/Chem/PandasTools.py [hunk 1]
```diff
@@ -648,12 +650,32 @@ def _runDoctests(verbose=None):  # pragma: nocover
 
 InstallPandasTools()
 
+try:
+  import xlsxwriter
+except ImportError:
+  xlsxwriter = None
+import unittest
```
[the remark this answers is not in the document]
Comment (greglandrum) [reply rdkit/Chem/PandasTools.py:657]: I would prefer to stick to standard python stuff for the tests themselves and just use pytest to find and run the tests. 

That way it remains possible to change test runners at some point in the future if we decide to do so.

Does pytest bring something enormously useful to the table that would make doing the change worth considering?

File: Projects/pytest.ini [hunk 2]
```diff
@@ -0,0 +1,4 @@
+[pytest]
+python_files = UnitTest*.py
+doctest_optionflags = ELLIPSIS NORMALIZE_WHITESPACE
+addopts =  --doctest-modules --ignore-glob='**/test_list.py'
```
[the remark this answers is not in the document]
Comment (greglandrum) [reply Projects/pytest.ini:4]: shouldn't be, I just forgot to update this.

File: rdkit/TestRunner.py
[the remark this answers is not in the document]
Comment (greglandrum) [reply rdkit/TestRunner.py:1]: yeah, it includes some code for redirecting stdout that's used in some of the tests

[the remark this answers is not in the document]
Comment (greglandrum) [reply rdkit/TestRunner.py:1]: > Is this file still used?

yeah, it includes some code for redirecting stdout that's used in some of the tests

**Comment by greglandrum** [comment]
> Have you tried running the tests on a build with `RDK_INSTALL_INTREE=OFF` ?

I did a build with `RDK_INSTALL_INTREE=OFF`, did:
```
make DESTDIR=/tmp/rdkit install
export PYTHONPATH=/tmp/rdkit/usr/local/lib/python3.10/site-packages
export LD_LIBRARY_PATH="/tmp/rdkit/usr/local/lib:/opt/boost_1_80/lib"
export RDBASE=/scratch/RDKit_git
```
and ran ctest from the build directory. There all the tests except `pythonSourceTests` pass. Those failures are due to ctest not having paths correct, but since I don't normally do this kind of build, I'm not sure there's a mistake in the way I set up the environment.

If I include `RDK_INSTALL_PYTHON_TESTS=ON` and repeat the above install command, then the python tests in `rdkit` are also copied with the python source files and .so files. After that, changing to `/tmp/rdkit/usr/local/lib:/opt/boost_1_80/lib` and running `python -m pytest` successfully finds and runs all the tests. So I think things are good.

File: rdkit/Chem/PandasTools.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ricrogz) [reply rdkit/Chem/PandasTools.py:657]: Well, pytest has a rather nice way to parametrize tests and set up fixtures, and it's also nice not to have to group tests under a class. Not that we have a huge need for any of that, but it's nice to have.

File: rdkit/TestRunner.py
[the remark this answers is not in the document]
Comment (ricrogz) [reply rdkit/TestRunner.py:1]: ok

**Comment by ricrogz** [comment]
> If I include `RDK_INSTALL_PYTHON_TESTS=ON` and repeat the above install command, then the python tests in `rdkit` are also copied with the python source files and .so files. After that, changing to `/tmp/rdkit/usr/local/lib:/opt/boost_1_80/lib` and running `python -m pytest` successfully finds and runs all the tests. So I think things are good.

Ah, I think I didn't enable that one. I'll give it a try later on.

**Review by ricrogz — APPROVED** [review]
Ok, I'm going to approve this PR. I still have some issues running tests on my computer, but I think these are probably due to my  own set up.

**Pull request merged**